frontend/src/app/main/ui/workspace/sidebar/options/menus/component.cljs

```diff
@@ -839,13 +838,14 @@
                                     :on-close on-menu-close
                                     :menu-entries menu-entries
                                     :main-instance true}]])]
-        [:*
-         (for [[pos property] (map vector (range) properties)]
-           (let [val (str/join ", " (:values property))]
-             [:div {:key (str (:id shape) (:name property)) :class (stl/css :variant-property-row)}
-              [:> input-with-values* {:name (:name property) :values val :on-blur (partial update-property-name pos)}]
-              [:> icon-button* {:variant "ghost"
-                                :aria-label (tr "workspace.shape.menu.remove-variant-property")
-                                :on-click (partial remove-property pos)
-                                :icon "remove"
-                                :disabled (<= (count properties) 1)}]]))]]])))
+        (when-not multi
+          [:*
+           (for [[pos property] (map vector (range) properties)]
```

Comment:
We already have the `(d/enumerate)` for this use case

Reply:
But instead of enumerating on each rerender, i think this should be handled when properties is defined.
Firstly, `properties` should be using `with-memo`
The index can be calcultated directly on that operation, so you don't need to iterate over it again for enumeration:

```clojure
(mf/with-memo [data objects variant-id]
  (->> (dwv/find-related-components data objects variant-id)
       (mapcat :variant-properties)
       (group-by :name)
       (map-indexed (fn [index [k v]]
                      {:name k
                       :pos index
                       :values (distinct
                                (map #(if (str/empty? (:value %)) "--" (:value %)) v))}))))
```
Son enumerating it here, just use `(:pos prop)` when you need position